From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 4 Jul 2018 15:30:22 -0400
Subject: [PATCH] Vex#get/setSummoner API

Get's the NPC that summoned this Vex and
Allow setting the vex's summoner

Co-authored-by: BillyGalbreath <Blake.Galbreath@GMail.com>

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftVex.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftVex.java
index fecb1e6b84fed241d1028ef24660804e5518ea3b..2ba16e33dd21c3c72cb12244aa78c59bf53e76d1 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftVex.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftVex.java
@@ -18,6 +18,19 @@ public class CraftVex extends CraftMonster implements Vex {
         return (net.minecraft.world.entity.monster.Vex) super.getHandle();
     }
 
+    // Paper start
+    @Override
+    public org.bukkit.entity.Mob getSummoner() {
+        net.minecraft.world.entity.Mob owner = getHandle().getOwner();
+        return owner != null ? (org.bukkit.entity.Mob) owner.getBukkitEntity() : null;
+    }
+
+    @Override
+    public void setSummoner(org.bukkit.entity.Mob summoner) {
+        getHandle().setOwner(summoner == null ? null : ((CraftMob) summoner).getHandle());
+    }
+    // Paper end
+
     @Override
     public String toString() {
         return "CraftVex";
